src/map/handler/Map.Drag.js

```diff
@@ -28,8 +28,9 @@ L.Map.Drag = L.Handler.extend({
 				dragend: this._onDragEnd
 			}, this);
 
+			this._draggable.on('predrag', this._onPreDragLimit, this);
```

Comment:
The map's maxBounds can be set dynamically, and it would be a real hassle to intercept changes to the maxBounds and attach/detach this event handler :-/